docs/ext/commands/additional_info.rst

```diff
@@ -32,3 +32,30 @@ If your bot requires shard distribution across several machines, you should set
 This will prevent conflicts and race conditions. Discord API doesn't provide users with events related to application command updates,
 so it's impossible to keep the cache of multiple machines synced. Having only 1 machine with ``sync_commands`` set to ``True`` is enough
 because global registration of application commands doesn't depend on sharding.
+
+.. _why_params_and_injections_return_any:
+
+Why ``Param`` and ``Injection``-related functions return ``Any``?
+-----------------------------------------------------------------
+
+If your editor of choice supports type-checking, you might have been noticed that :func:`~ext.commands.Param`, :func:`~ext.commands.inject`,
+:func:`~ext.commands.injection` and :func:`~ext.commands.register_injection` return ``Any`` (in terms of typing), but at runtime
+return :class:`~ext.commands.Injection` (as mentioned in API reference for them). "Why?", you could think, and here's the reason for this.
+
+We'll explain this on example of ``Param`` usage (because this is the thing most users are familair with), but everything here applies to
+``Injection`` usage as well.
+
```

Comment:
```suggestion
```

Reply:
Shortened much as of latest commit, won't resolve this for now